python_app/Dockerfile

```diff
@@ -1,6 +1,11 @@
 FROM python:3.5
-RUN pip install flask; pip install python-consul; pip install docker; pip install redis
-ADD ./ /src
+
 WORKDIR /src
-EXPOSE 5000
-ENTRYPOINT python app.py
+
+COPY requirements.txt .
+
+RUN pip install -r requirements.txt
+
+COPY ./ .
```

Comment:
isn't requirements.txt part of this?  why don't we just copy it all after WORKDIR and remove requirements.txt?  (unless i'm wrong)

Reply:
Yeah, mi bad, a pip install -f shoud do the trick

Reply:
> isn't requirements.txt part of this? why don't we just copy it all after WORKDIR and remove requirements.txt? (unless i'm wrong)

I divided on 2 parts, so we can have a better cache handling. The requirements usually are dependencies that don't change a lot and are quite heavy, so it could be better to have it at the top and have the source code is more susceptible to change, so if it's in a lower layer we can save time on compiling

Reply:
I like the way you're thinking.  Agreed - move on.